compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -573,13 +573,31 @@ impl AcirContext {
         // lhs = rhs * q + r
         //
         // If predicate is zero, `q_witness` and `r_witness` will be 0
+        let zero = self.add_constant(FieldElement::zero());
+        if self.var_to_expression(predicate)?.is_zero() {
+            return Ok((zero, zero));
+        }
+
+        // If `lhs` and `rhs` are known constants then we can calculate the result at compile time.
+        if let (Some(lhs_const), Some(rhs_const)) =
+            (self.var_to_expression(lhs)?.to_const(), self.var_to_expression(rhs)?.to_const())
+        {
+            // Disallow division by zero.
+            if rhs_const != FieldElement::zero() {
+                let quotient = lhs_const.to_u128() / rhs_const.to_u128();
+                let remainder = lhs_const.to_u128() - quotient * rhs_const.to_u128();
+
+                let quotient_var = self.add_constant(FieldElement::from(quotient));
+                let remainder_var = self.add_constant(FieldElement::from(remainder));
+                return Ok((quotient_var, remainder_var));
```

Comment:
Just checking... Do we want to multiply by the predicate here to enforce consistency between this and the full implementation? For example, currently we have:

```
// If all inputs are only known at runtime, we then get the full expression
euclidean_division_var(lhs, rhs, p) -> if p == 0 { (0,0) } else { (p/q, p%q) };

// If `lhs` and `rhs` are known at compile-time we instead simplify down to the result.
euclidean_division_var(5, 3, p) -> (1, 2);
```

We're then getting a different result in the `p==0` case depending on whether the inputs are known at compile-time. Seems like it shouldn't cause an issue however.